### ISSUE TITLE
[Android] SymbolLayer style causing instant crash

### ISSUE THREAD

**Issue by 2f-concept** [react-native-mapbox-gl/maps#283]
**Describe the bug**
Trying to set iconImage from features properties cause a java.lang.ClassCastException that crash the application.
Here are the logs :
```
java.lang.ClassCastException: Value for value cannot be cast from ReadableNativeArray to String
        at com.facebook.react.bridge.ReadableNativeMap.checkInstance(ReadableNativeMap.java:150)
        at com.facebook.react.bridge.ReadableNativeMap.getNullableValue(ReadableNativeMap.java:142)
        at com.facebook.react.bridge.ReadableNativeMap.getString(ReadableNativeMap.java:192)
        at com.mapbox.rctmgl.components.styles.RCTMGLStyleValue.<init>(RCTMGLStyleValue.java:49)
        at com.mapbox.rctmgl.components.styles.RCTMGLStyle.getStyleValueForKey(RCTMGLStyle.java:59)
        at com.mapbox.rctmgl.components.styles.RCTMGLStyleFactory.setSymbolLayerStyle(RCTMGLStyleFactory.java:188)
        at com.mapbox.rctmgl.components.styles.layers.RCTMGLSymbolLayer.addStyles(RCTMGLSymbolLayer.java:45)
        at com.mapbox.rctmgl.components.styles.layers.RCTLayer.addToMap(RCTLayer.java:246)
        at com.mapbox.rctmgl.components.styles.layers.RCTMGLSymbolLayer.addToMap(RCTMGLSymbolLayer.java:29)
        at com.mapbox.rctmgl.components.styles.sources.RCTSource.addLayerToMap(RCTSource.java:194)
        at com.mapbox.rctmgl.components.styles.sources.RCTSource$1.onStyleLoaded(RCTSource.java:129)
        at com.mapbox.mapboxsdk.maps.MapboxMap.getStyle(MapboxMap.java:106)
        at com.mapbox.rctmgl.components.styles.sources.RCTSource.addToMap(RCTSource.java:117)
        at com.mapbox.rctmgl.components.styles.sources.RCTMGLShapeSource.access$401(RCTMGLShapeSource.java:32)
        at com.mapbox.rctmgl.components.styles.sources.RCTMGLShapeSource$1.onStyleLoaded(RCTMGLShapeSource.java:69)
        at com.mapbox.mapboxsdk.maps.MapboxMap.notifyStyleLoaded(MapboxMap.java:852)
        at com.mapbox.mapboxsdk.maps.MapboxMap.onFinishLoadingStyle(MapboxMap.java:208)
        at com.mapbox.mapboxsdk.maps.MapView$MapCallback.onDidFinishLoadingStyle(MapView.java:1215)
        at com.mapbox.mapboxsdk.maps.MapChangeReceiver.onDidFinishLoadingStyle(MapChangeReceiver.java:196)
        at com.mapbox.mapboxsdk.maps.NativeMapView.onDidFinishLoadingStyle(NativeMapView.java:1035)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:323)
        at android.os.Looper.loop(Looper.java:136)
        at android.app.ActivityThread.main(ActivityThread.java:6077)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756)
```

**To Reproduce**
Here is my code:
```
const markerStyles = {
  icon: {
    iconImage: ['get', 'icon'],
    iconSize: 1
  },
};

const markers = {
    type    : "FeatureCollection",
    features: [
      {
        type: "Feature",
        properties: {
          objectId: "S9FIc2gBHnoQSG7pDdbj",
          ttl: "Pizza A Emporter",
          icon: "airport-15"
        },
        geometry: {
          type: "Point",
          coordinates: [
            4.837938,
            45.791937
          ]
        },
      },
      {
        type: "Feature",
        properties: {
          objectId: "yM1nb2gBHnoQSG7pt6XF",
          ttl: "Pharmacie du Radiant",
          icon: "airport-15"
        },
        geometry: {
          type: "Point",
          coordinates: [
            4.837938,
            45.791937
          ]
        },
      }
    ]
}
...
renderMap() {
    const { mapZoom, mapCenter } = this.state;
    const { mapboxStyleURL } = appConfig;

    return (
      <MapboxGL.MapView style={{flex: 1}}
        styleURL={mapboxStyleURL}
        attributionEnabled={false}
      >
        {mapZoom && mapCenter && (
          <MapboxGL.Camera
            zoomLevel={17}
            centerCoordinate={mapCenter}
            rotateEnabled={false}
          />
        )}

        <MapboxGL.ShapeSource
          id="symbolLocationSource"
          hitbox={{width: 20, height: 20}}
          rotateEnabled={false}
          shape={markers}
          onPress={(e) => this.showObject(e.nativeEvent.payload.properties.objectId)}
        >
          <MapboxGL.SymbolLayer
            id="symbolLocationIcons"
            minZoomLevel={1}
            style={markerStyles.icon}
          />

        </MapboxGL.ShapeSource>
      </MapboxGL.MapView>
    )
  }
```

**Expected behavior**
SymbolLayer showing icon according to feature property

**Versions :**
 - Platfrom: Android
 - Device: Galaxy Nexus virtual device (emulator)
 - OS: tryed with android 7.0 and 8.1
 - SDK Version: 28
 - React Native Version: 0.59.3

**Comment by 2f-concept** [comment]
I can make it work if I set un map package dependencies :
```
...
"@react-native-mapbox-gl/maps": "file:../react-native-mapbox-gl-maps.tgz",
...
```
Anyone could explain me why ?

**Comment by d3vhound** [comment]
Yeah this shouldn't be recommended over PointAnnotation if it doesn't work with custom images

**Comment by mfazekas** [comment]
Not mentioned in the report which rnmpgl versions are we talking about, please try master, i think this is fixed by #124

**Comment by d3vhound** [comment]
is #124 apart of npm version 7.0.0-rc3 ? if so it's not fixed or documentation is lacking some important steps.
